src/compiler/scanner.ts

```diff
@@ -1361,7 +1361,9 @@ namespace ts {
                         if (text.charCodeAt(pos + 1) === CharacterCodes.equals) {
                             return pos += 2, token = SyntaxKind.LessThanEqualsToken;
                         }
-                        if (text.charCodeAt(pos + 1) === CharacterCodes.slash && languageVariant === LanguageVariant.JSX) {
+                        if (text.charCodeAt(pos + 1) === CharacterCodes.slash &&
+                                text.charCodeAt(pos + 2) !== CharacterCodes.asterisk &&
+                                languageVariant === LanguageVariant.JSX) {

```

Comment:
i would move the `languageVariant === LanguageVariant.JSX` to the beginning of the if condition.